### PR TITLE
Upgrade MaxText training CI and dependencies to JAX 0.11.1

### PR DESCRIPTION
**Pull request by darisoy**
# Description

Upgrade MaxText training CI and pinned dependencies to **JAX 0.11.1** (build commit [`ab4c9b943c70bcb42baf4d379036a19c8aa2689d`](https://github.com/jax-ml/jax/commit/ab4c9b943c70bcb42baf4d379036a19c8aa2689d)).

This is needed to support newer `jax.experimental.layout` and `must_fuse_call` features used in recent flash attention kernel optimizations (cl/952800018 / b/552049890).

* Regenerated `src/dependencies/requirements/generated_requirements/tpu-requirements.txt`
* Regenerated `src/dependencies/requirements/generated_requirements/cuda12-requirements.txt`
* Regenerated `src/dependencies/requirements/generated_requirements/decoupled-requirements.txt`
* Updated JAX build commit hashes in `docs/development/update_dependencies.md`

# Tests
* Verified on TPU v6e-4 VM (`darisoy-gvnic-test`):
  * `pytest tests/unit/attention_test.py` (76 passed, 72 skipped, 0 failed)
  * `pytest tests/unit/linears_test.py tests/unit/llama_test.py tests/unit/embeddings_test.py tests/unit/optimizers_test.py` (11 passed, 51 skipped, 0 failed)

FIXES: b/552049890

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/dependencies/requirements/generated_requirements/decoupled-requirements.txt [hunk 1]
```diff
@@ -110,116 +114,116 @@ namex>=0.1.0
...
 networkx>=3.6.1
 ninja>=1.13.0
-nltk>=3.9.4
+nltk>=3.10.3
 nodeenv>=1.10.0
-numpy>=2.0.2
-numpy-typing-compat>=20251206.2.0
+numpy>=2.1.3
+numpy-typing-compat>=20260602.2.1
+nvidia-cudnn-frontend>=1.27.0
```
[the remark this answers is not in the document]
Comment (darisoy) [reply src/dependencies/requirements/generated_requirements/decoupled-requirements.txt]: Resolved. Added `nvidia-cudnn-` to `EXCLUDED_ACCELERATOR_PREFIXES` in `generate_decoupled_requirements.py` and regenerated `decoupled-requirements.txt`.

**Review by gobbleturk — APPROVED** [review]

**Review by bvandermoon — APPROVED** [review]
LGTM, thank you @darisoy. Doesn't need to be this PR, but should we also update the post-training JAX version?

**Review by SurbhiJainUSC — APPROVED** [review]
